src/controllers/csi/provisioner/controller.go

```diff
@@ -101,69 +101,102 @@ func (provisioner *OneAgentProvisioner) Reconcile(ctx context.Context, request r
+	latestProcessModuleConfigCache, requeue, err := provisioner.updateProcessModuleConfigCache(dtc, dynakubeMetadata, dk)
+	if requeue {
+		return reconcile.Result{RequeueAfter: defaultRequeueDuration}, err
+	}
 	if err != nil {
-		log.Error(err, "error when getting the latest ruxitagentproc.conf")
 		return reconcile.Result{}, err
 	}
+
+	// Set/Update the `LatestVersion` field in the database entry
+	err = provisioner.createOrUpdateDynakubeMetadata(oldDynakubeMetadata, dynakubeMetadata)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	err = provisioner.writeProcessModuleConfigCache(dynakubeMetadata.TenantUUID, latestProcessModuleConfigCache)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{RequeueAfter: defaultRequeueDuration}, nil
+}
+
+func (provisioner *OneAgentProvisioner) updateProcessModuleConfigCache(dtc dtclient.Client, dynakubeMetadata *metadata.Dynakube, dk *dynatracev1beta1.DynaKube) (
+	latestProcessModuleConfigCache *processModuleConfigCache,
+	requeue bool,
+	err error,
+) {
+	latestProcessModuleConfig, storedHash, err := provisioner.getProcessModuleConfig(dtc, dynakubeMetadata.TenantUUID)
+	if err != nil {
+		log.Error(err, "error when getting the latest ruxitagentproc.conf")
+		return nil, false, err
+	}
 	latestProcessModuleConfig = latestProcessModuleConfig.AddHostGroup(dk.HostGroup())
-	latestProcessModuleConfigCache := newProcessModuleConfigCache(latestProcessModuleConfig)
+
+	connectionInfo, err := dtc.GetConnectionInfo()
+	if err != nil {
+		log.Error(err, "error when getting OneAgent connectionInfo")
+		return nil, false, err
+	}
+	latestProcessModuleConfig = latestProcessModuleConfig.AddConnectionInfo(connectionInfo)
+
+	latestProcessModuleConfigCache = newProcessModuleConfigCache(latestProcessModuleConfig)
 
 	agentUpdater := newAgentUpdater(dtc, provisioner.path, provisioner.fs, provisioner.recorder, dk)
-	if updatedVersion, err := agentUpdater.updateAgent(dynakube.LatestVersion, dynakube.TenantUUID, storedHash, latestProcessModuleConfigCache); err != nil {
+	if updatedVersion, err := agentUpdater.updateAgent(dynakubeMetadata.LatestVersion, dynakubeMetadata.TenantUUID, storedHash, latestProcessModuleConfigCache); err != nil {
 		log.Info("error when updating agent", "error", err.Error())
 		// reporting error but not returning it to avoid immediate requeue and subsequently calling the API every few seconds
-		return reconcile.Result{RequeueAfter: defaultRequeueDuration}, nil
+		return nil, true, nil
 	} else if updatedVersion != "" {
-		dynakube.LatestVersion = updatedVersion
+		dynakubeMetadata.LatestVersion = updatedVersion
 	}
+	return latestProcessModuleConfigCache, false, nil
+}
 
-	// Set/Update the `LatestVersion` field in the database entry
-	err = provisioner.createOrUpdateDynakube(oldDynakube, dynakube)
+func (provisioner *OneAgentProvisioner) handleMetadata(dk *dynatracev1beta1.DynaKube) (*metadata.Dynakube, metadata.Dynakube, reconcile.Result, error) {
```

Comment:
Would remove reconcile.Result from return values as it is always the same empty struct